### DISCUSSION TITLE
Object Browser filter - when hovering over a source file, reported info is sometimes incorrect

### DISCUSSION THREAD

**Discussion by heymchri** [codefori/vscode-ibmi#2708] [Q&A]
![image](https://github.com/user-attachments/assets/180f0c38-9069-40ac-930f-396cd580449c)
When hovering over a source file, if the file has a CCSID of 65535, the reported CCSID shows as 37.
Also, if a member was added outside of VS Code, upon taking the refresh option on the source file, the member count is not updated. It is only updated if the refresh option is taken on the filter.
Also, although the Source File CCSID setting on the connection configuration is set to the default value *FILE, the source member opens just fine, although that should result in an error as source files with CCSID 65535 are supposedly not supported (unless the Source File CCSID is changed to a proper CCSID value).

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.13.5|
|Visual Studio Code version|1.94.2|
|Operating System|darwin_arm64|

<details>
<summary>Active extensions</summary>

```
.NET Install Tool (vscode-dotnet-runtime): 2.2.0
CL (vscode-clle): 1.1.7
COBOL (cobol): 24.10.15
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.0
Dev Containers (remote-containers): 0.388.0
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.20.0
Excel Viewer (gc-excelviewer): 4.2.62
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
IBM i Debug (ibmidebug): 2.0.1
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Overtype (overtype): 0.5.0
Prettier - Code formatter (prettier-vscode): 11.0.0
Print (vscode-print): 0.13.2
SQL Server (mssql) (mssql): 1.24.0
Server Ready Action (debug-server-ready): 1.0.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
vscode-icons (vscode-icons): 12.9.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|10|
|CCSID Origin|65535|
|Runtime CCSID|65535|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Enabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|/QOpenSys/QIBM/ProdData/JavaVM/jdk11|
|-|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|64bit|
|chsh|iconv||||||
|ls|setccsid||||||
|md5sum|tar||||||
|sort|||||||
|stat|||||||
|tn5250|||||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=HEYMCHR
CURLIB=HEYMCHR
HOME=/home/HEYMCHR
HOST=usalid06.infor.com
LIBLS=HEYMCHR QTEMP QGPL
LOGIN=heymchr
LOGNAME=heymchr
MAIL=/var/spool/mail/heymchr
OLDPWD=/home/heymchr
PATH=/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/HEYMCHR
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=10.61.0.150 59459 22
SSH_CONNECTION=10.61.0.150 59459 10.39.80.55 22
TZ=<EST>5<EDT>,M3.2.0,M11.1.0
USER=heymchr
USERNAME=heymchr
WORKDIR=/home/HEYMCHR
_=/QOpenSys/pkgs/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF9801:  Object QCPFRMIMPF in library QSYS not found.",
    "cwd": "/home/HEYMCHR"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF3012:  File TRNPTFP1 in library *LIBL not found.",
    "cwd": "/home/HEYMCHR"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF3012:  File TRNPTFP1 in library *LIBL not found.",
    "cwd": "/home/HEYMCHR"
  }
]
```
</details>

**Comment by worksofliam** [comment]
Please share the contents of your Code for IBM i output channel after hovering over a source file and seeing the incorrect info.

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3708366/178739532-03c8ad30-84de-4698-9904-65938f2ef651.png">

**Comment by heymchri** [comment]
Hi Liam,
see attached text file.
Thx,
Christophe
[output.txt](https://github.com/user-attachments/files/17439600/output.txt)

**Comment by worksofliam** [comment]
@heymchri Well, the system is returning 37, and not 65535, to us:

```
/home/HEYMCHR: /usr/bin/attr -p /QSYS.LIB/VSCODETEST.LIB/QSRC65535.FILE CCSID
{
    "code": 0,
    "signal": null,
    "stdout": "CCSID=37",
    "stderr": ""
}
```

Are you able to do a `DSPFFD` on this table and paste the entire output here?

**Comment by heymchri** [comment]
Hi @worksofliam here you go:

[QPDSPFFD_HEYMCHR_QPADEV001G_045759_1.pdf](https://github.com/user-attachments/files/18931647/QPDSPFFD_HEYMCHR_QPADEV001G_045759_1.pdf)

**Comment by worksofliam** [comment]
@heymchri Looked into this more, and there isn't much more we can do since we are reading directly from the OS:

* `/usr/bin/attr` is returning 37 in your output log.
* `DSPFFD` is returning 65535 as defined.

I don't believe this to be a bug on our side, so I am going to convert this to a discussion.

Is CCSID 37 used anywhere else in your system? What is your system CCSID, locale, etc? What is your job job CCSID and default CCSID?

**Comment by heymchri** [comment]
@worksofliam QCCSID is 37, local is EN_US, job CCSID is 37, default CCSID is 37.
I guess I should be able to report this to IBM as a bug with attr, since it's clearly returning the wrong CCSID?
